code/src/App.js

```diff
@@ -1,9 +1,72 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react'
+import { ListOfThoughts } from 'components/ListOfThoughts'
+import { NewThoughtPost } from 'components/NewThoughtPost'
 
 export const App = () => {
+  const [thoughtsList, setThoughtsList] = useState([])
+  const [newThought, setNewThought] = useState('')
+  const [loading, setLoading] = useState(false)
+
+  const fetchThoughts = () => {
+    setLoading(true)
+    fetch('https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts')
+      .then((res) => res.json())
+      .then((data) => setThoughtsList(data))
+      .catch((error) => console.error(error))
+      .finally(() => setLoading(false))
+  }
+
+  useEffect(() => {
+    fetchThoughts()
+  }, [setNewThought])
+
+  const handleNewThoughtChange = (event) => {
+    setNewThought(event.target.value)
+  }
+
+  const onFormSubmit = (event) => {
+    event.preventDefault()
+
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        message: newThought
+      })
+    }
+
+    fetch('https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts', options)
+      .then((res) => res.json())
+      .then(() => fetchThoughts())
+      .finally(() => setNewThought(''))
+  }
+
```

Comment:
Given that this project was quite straightforward, I think it was wise to only have two components and implementing the fetching and other functions in App main component.